pkg/backend/endpoint_resolver.go

```diff
@@ -2,15 +2,19 @@ package backend
 
 import (
 	"context"
+	"fmt"
 	"github.com/go-logr/logr"
 	"github.com/pkg/errors"
 	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/apimachinery/pkg/util/intstr"
 	"sigs.k8s.io/aws-load-balancer-controller/pkg/k8s"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+var ErrNotFound = errors.New("backend not found")
```

Comment:
nit:
The error message contains redundant "not found" phrase . For example,
"backend not found: services \"svc-1\" not found"

I am more inclined to have something like - 
"backend: services \"svc-1\" not found"
or
"backend error: services \"svc-1\" not found"